checks/cloud/aws/iam/no_policy_wildcards.go

```diff
@@ -165,3 +159,21 @@ func checkStatement(src iam.Document, statement iamgo.Statement, results scan.Re
 
 	return results
 }
+
+func isObjectKeyContainsWildcard(arn string) bool {
+	// https://docs.aws.amazon.com/IAM/latest/UserGuide/reference-arns.html
+	// arn:partition:service:region:account-id:resource-id
+	arnParts := strings.SplitN(arn, ":", 6)
+
+	if len(arnParts) != 6 || arnParts[2] != "s3" {
+		return false
+	}
+
+	resourceParts := strings.SplitN(arnParts[5], "/", 2)
+	if len(resourceParts) != 2 {
+		return false
+	}
+
+	return !strings.ContainsRune(resourceParts[0], '*') && strings.ContainsRune(resourceParts[1], '*')
```

Comment:
this logic seems a little brittle to me, don't you think?

Reply:
I don't think the [ARN](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference-arns.html) format will change. Does it make sense to use the [aws-sdk-go](https://github.com/aws/aws-sdk-go-v2/blob/main/aws/arn/arn.go) package for such a simple check?

Reply:
Fair enough